marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/CallBatcherImpl.java

```diff
@@ -287,6 +293,8 @@ public CallBatcher add(W input) {
 
         CallManagerImpl.CallArgsImpl args = null;
         if (input == null) {
+            if(inputType == Void.class)
+                throw new IllegalArgumentException("Input type is void");
```

Comment:
A more useful error message might be something like:  "cannot call add() when supplying arguments with generator"